### ISSUE TITLE
Invalid program exception

### ISSUE THREAD

**Issue by kemsky** [dadhi/FastExpressionCompiler#357]
I've created sample project: https://github.com/kemsky/fast-compile-bug

```
Unhandled exception. System.InvalidProgramException: Common Language Runtime detected an invalid program.
   at (ArrayClosure , ActionItem )
   at Program.<Main>$(String[] args)

```

Looks like this is somehow related to interface default methods or inner classes.

**Comment by dadhi** [comment]
Good start. 
Could you extract a smaller failing test from it?

**Comment by kemsky** [comment]
@dadhi, I remove most of the irrelevant code. The only suspicious part is operator overload.

**Comment by dadhi** [comment]
Ok, seems legit. Operator overloading is not something I have tested FEC suffiently with.
I think, if you replace operators with method calls then you may confirm whether it is the root cause.

**Comment by kemsky** [comment]
I'm not sure what else I can do because expression is aready as simple as it can get.

**Comment by dadhi** [comment]
@kemsky Huh, thanks. 
I did not check your repo after the message. 
Will check the issue when I have time.

**Comment by kemsky** [comment]
@dadhi, could you publish this fix?

**Comment by dadhi** [comment]
@kemsky There is one thing I am working on for release. I will check if it possible to release just the fix.

**Comment by dadhi** [comment]
@kemsky Sorry for the late response, I have released the v3.4.0-preview-01 with the fix right now, and planning to release RTM soon.
